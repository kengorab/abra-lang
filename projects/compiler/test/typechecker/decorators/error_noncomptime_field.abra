decorator Bar {
  p: Person
}

type Person { name: String }
