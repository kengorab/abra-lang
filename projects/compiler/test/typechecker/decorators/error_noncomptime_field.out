Error at %TEST_DIR%/typechecker/decorators/error_noncomptime_field.abra:2:3
Forbidden field type for decorator 'Bar'
  |    p: Person
       ^
This field has type 'Person', which is not allowed for fields of decorators.
Decorators' fields must be one of: Int, Float, Bool, String
